src/api/mod.rs

```diff
@@ -41,6 +46,7 @@ impl<'r, 'o: 'r> Responder<'r, 'o> for ApiError {
     fn respond_to(self, _request: &'r Request<'_>) -> rocket::response::Result<'o> {
         error!("API error: {}", self);
         let (status, message) = match self {
+            ApiError::Generic(ref err) => (Status::InternalServerError, err.to_string()),
```

Comment:
note:
`default_catcher` wraps an `ApiError::Generic` inside `response::status::Custom`.
Inside `Responder` we always map `Generic` → `Status::InternalServerError`, but the outer `Custom` will overwrite the HTTP status with the actual status that triggered the catcher (404, 401, …).
The response body will therefore claim `"status":500` while the header is, say, `404 Not Found`.

You could pass the original status into the error variant - i.e.:
`Generic(Status, String)` - and adjust the Responder

In such a scenario you may need to update the `default_catcher` 0 ==- e.g.
```rust
fn default_catcher(...)
    ...
    response::status::Custom(
        status,
        ApiError::Generic(status, format!("{} - {}", status.code, status.reason_lossy())),
    )
```

Reply:
done!